tools/test/h5jam/getub.c

```diff
@@ -138,14 +138,15 @@ main(int argc, const char *argv[])
     } /* end if */
 
     /* close things and exit */
+    HDfree(filename);
     HDfree(buf);
     HDclose(fd);
 
     return EXIT_SUCCESS;
 
 error:
-    if (buf)
-        HDfree(buf);
+    HDfree(filename);
+    HDfree(buf);
```

Comment:
HDfree doesn't protect against NULL pointers, so it would be best to leave those in for the error situations.

Reply:
The C standard says that freeing a NULL pointer is fine so checking for NULL isn't necessary.